performancecopilot/pcp
# assorted qa Changes committed to git@github.com:kmcdonell/pcp.git 20190904

Ken McDonell (9):
      qa/967.out: oops, cut-n-paste error in header line from qa/1100
      qa/common.check: add check for defunct process in _wait_pmlogger_end()
      qa/660: remove more prattle
      qa/870: report current datestamp with --check and errors
      qa/555: accommodate RHEL 7.x (CentOS 7.x) oddities
      qa/555: similar changes to the last ones
      qa/555: more junk filtering
      qa/1141: notrun if pcpupstream module not loaded
      qa/590: notrun for openbsd

 qa/1141         |   13 ++++++++++---
 qa/555          |   34 +++++++++++++++++++++++-----------
 qa/590          |   15 +++++++++++++++
 qa/660          |    2 ++
 qa/870          |    6 +++++-
 qa/967.out      |    2 +-
 qa/common.check |   19 ++++++++++++++++++-
 7 files changed, 74 insertions(+), 17 deletions(-)

Details ...

commit 75c2bf662d9dd0d1d0b762d75a29c73f94aadb3f
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Fri Sep 6 14:24:35 2019 +1000

    qa/590: notrun for openbsd
    
    From the comment added in this commit ...
        This is a bad combination ... we're not sure if it is libc
        on openbsd where fopen() is not thread safe leading to:
        pmNewContext(archives/ok-foo) -> No such file or directory
        or our implementation of "thread private data" that is busted
        with clang, but as this test passes everywhere else, we've opted
        to "notrun" it here.

commit 561c1b383c860314d2d9056a1923e4c581dea937
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Fri Sep 6 08:53:38 2019 +1000

    qa/1141: notrun if pcpupstream module not loaded
    
    If we're on a system with selinux tools installed, but the pcpupstream
    module is not loaded, then we run the risk of this test failing because
    some other selinux dependency is not satisfied.  For example on SUSE
    SLES12 (vm30), I'm seeing this:
    
    Failed to resolve typeattributeset statement at /var/lib/selinux/minimum/tmp/modules/400/pcpupstream/cil:1
    semodule:  Failed!
    
    So if we can't load the official pcpupstream module, then there is no
    hope for qa/1141, so notrun it.

commit 14dd939544b1c458b3791d954ada764da18c821c
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Thu Sep 5 15:43:38 2019 +1000

    qa/555: more junk filtering
    
    syslog(3) on RHEL 7.x (CentOS 7.x) is putting junk into syslog in our
    error test cases (which are rejected on other platforms), so make
    the qa test filter even more draconian ... only look for the stuff
    we expect to be there and ignore everything else.

commit f1a0cd07a05ea1011e37223b7c99687d3305b0e4
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Thu Sep 5 12:43:11 2019 +1000

    qa/555: similar changes to the last ones
    
    This time for RHEL 7.6 (vm29).

commit f48801be871847dccaab4b4c9a3ac65d078cb96c
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Thu Sep 5 11:46:23 2019 +1000

    qa/555: accommodate RHEL 7.x (CentOS 7.x) oddities
    
    1. changed syslog(3) semantics, so once case produces different
       output
    2. trashing of syslog (may be a different version of 1.)
    3. chatter in syslog
    
    Deal with all of this in the filtering.

commit c415e88be8d972c1b8b70f104677663aa4383285
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Thu Sep 5 09:20:46 2019 +1000

    qa/870: report current datestamp with --check and errors
    
    Helps diagnose if files reported are old or new.

commit b979a2db71f45a1fcff441f77a41c7a5b351a7cd
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Thu Sep 5 09:19:28 2019 +1000

    qa/660: remove more prattle
    
    warnings from deep inside the libraries pmweb uses.

commit 4de2ef16cb98b84fbb577baf18af328ebaec5382
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Thu Sep 5 09:00:02 2019 +1000

    qa/common.check: add check for defunct process in _wait_pmlogger_end()
    
    pmlogger really has exited if it is defunct.
    
    Seen in qa/1146 on some slow single CPU VMs.

commit 1a9d05cdaa405e43517dffbb2db4484352a2b8e0
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Wed Sep 4 16:50:31 2019 +1000

    qa/967.out: oops, cut-n-paste error in header line from qa/1100
